recipes/openssl/1.x.x/test_package/conanfile.py

```diff
@@ -22,21 +24,32 @@ def _skip_test(self):
         return self.settings.os == "Macos" and self.settings.arch == "armv8" \
                and self.options["openssl"].shared
 
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if self.settings.os == "Android":
+            tc.cache_variables["CONAN_LIBCXX"] = ""
+        openssl_version = Version(self.deps_cpp_info["openssl"].version)
+        if openssl_version.major == "1" and openssl_version.minor == "1":
+            tc.cache_variables["OPENSSL_WITH_ZLIB"] = False
+        else:
+            tc.cache_variables["OPENSSL_WITH_ZLIB"] = not self.options["openssl"].no_zlib
```

Comment:
```suggestion
        openssl = self.dependencies["openssl"]
        openssl_version = Version(openssl.ref.version)
        if openssl_version.major == "1" and openssl_version.minor == "1":
            tc.cache_variables["OPENSSL_WITH_ZLIB"] = False
        else:
            tc.cache_variables["OPENSSL_WITH_ZLIB"] = not openssl.options.no_zlib
```

docs.conan.io/en/latest/reference/conanfile/dependencies.html